cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_kernel_impl.i

```diff
@@ -41,22 +41,24 @@ namespace internal
 template <typename algorithmFPType, CpuType cpu>
 services::Status KNNClassificationPredictKernel<algorithmFPType, cpu>::compute(const NumericTable * data, const classifier::Model * m,
                                                                                NumericTable * label, NumericTable * indices, NumericTable * distances,
-                                                                               const daal::algorithms::Parameter * par)
+                                                                               const KernelParameter * par)
 {
     const Model * const convModel        = static_cast<const Model *>(m);
     NumericTableConstPtr trainDataTable  = convModel->impl()->getData();
     NumericTableConstPtr trainLabelTable = convModel->impl()->getLabels();
 
-    const Parameter * const parameter   = static_cast<const Parameter *>(par);
-    const uint32_t k                    = parameter->k;
-    const uint32_t nClasses             = parameter->nClasses;
-    const VoteWeights voteWeights       = parameter->voteWeights;
-    const DAAL_UINT64 resultsToEvaluate = parameter->resultsToEvaluate;
-    const DAAL_UINT64 resultsToCompute  = parameter->resultsToCompute;
+    const KernelParameter * const parameter     = static_cast<const KernelParameter *>(par);
```

Comment:
Looks like there is no reason to do cast in this context since function argument already has this type

Reply:
removed this, thx